test/object-store/realm.cpp

```diff
@@ -3601,3 +3601,25 @@ TEST_CASE("KeyPathMapping generation") {
         REQUIRE(q.count() == 0);
     }
 }
+
+TEST_CASE("collections with non nullable mixed should throw") {
```

Comment:
This test case isn't really adding anything. We could test schema validation at every single point that can produce a schema validation error, but it just duplicates a lot of testing for no benefit.

Reply:
Fair point. I started with this test and it does repro the assertion failure while the other tests do not, but having both does seem redundant now.